### PR TITLE
Change Report category to "violation" if rule IDs are provided

### LINKED ISSUE #20120: API: reporting an account fails if `rule_ids[]` is provided but `category` is not set to `violation`
### Steps to reproduce the problem

1. `POST http://mastodon.local/api/v1/reports?account_id=109307959913649155&status_ids[]=109308029266383624&rule_ids[]=1`

### Expected behaviour

200 OK, Report is successful

### Actual behaviour

422 Validation failed: Rule ids must be blank

### Detailed description

bit weird that `/api/v1/reports` expects `rule_ids[]` to be blank when the `category` is something other than `violation` -- perhaps the response can be more graceful and robust?

maybe if `category=violation` is not provided, then `rule_ids[]` is ignored? or better, if `rule_ids[]` is present, then the `category` is changed to `violation`?

### Specifications

4.0.0rc2

### PR DESCRIPTION
**Pull request by trwnh**
Fix #20120

I am not sure if there is something else I should be doing (perhaps related to options.delete) but this at least works with the intended result. If there is a more idiomatic way of expressing this, then let me know in a code review?

### CODE REVIEW

File: app/services/report_service.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ def call(source_account, target_account, options = {})
     @target_account = target_account
     @status_ids     = options.delete(:status_ids).presence || []
     @comment        = options.delete(:comment).presence || ''
-    @category       = options.delete(:category).presence || 'other'
+    @category       = 'violation' if options[:rule_ids].present? || options.delete(:category).presence || 'other'
```
Comment (Gargron) [inline app/services/report_service.rb:11]: This line will only set `@category`, and only to `violation`, when either rule IDs are present, or category is present, or when string `other`. That is not your intent.

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/services/report_service.rb (hunk 1 above)
Comment (trwnh) [reply app/services/report_service.rb:11]:
```ruby
options[:rule_ids].present? ? 'violation' : options.delete(:category).presence || 'other'
```

Would this work?

Comment (Gargron) [reply app/services/report_service.rb:11]: You need paranthesis around the last part

Comment (trwnh) [reply app/services/report_service.rb]: Seems to not be necessary but I added them anyway just now

Comment (trwnh) [reply app/services/report_service.rb]: Actually adding parentheses *breaks* the code. Adding rule_ids[] makes the category "other" when parenthesized

**Review by Gargron — APPROVED** [review]

**Pull request merged**